lib/facility.rb

```diff
@@ -1,14 +1,65 @@
+require './lib/dmv'
+require './lib/facility'
+require './lib/vehicle'
+require './lib/dmv_data_service'
+
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+    @address = facility_details[:address]
+    @collected_fees = 0
+    @name = facility_details[:name]
+    @phone = facility_details[:phone]
+    @registered_vehicles = []
+    @services = [] 
+    
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  
+  def register_vehicle(vehicle)
+    vehicle.change_plate
+    vehicle.change_date
```

Comment:
I like that you put these methods in the vehicle class. That seems like the responsibility of the Vehicle class.